samples/durable-task-sdks/dotnet/Agents/PromptChaining/Worker/Program.cs

```diff
@@ -0,0 +1,139 @@
+using Azure.Identity;
+using Microsoft.DurableTask;
+using Microsoft.DurableTask.Worker;
+using Microsoft.DurableTask.Worker.AzureManaged;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using AgentChainingSample.Services;
+using AgentChainingSample.Activities;
+using AgentChainingSample.Orchestrations;
+using AgentChainingSample.Shared.Models;
+
+// Configure the host builder
+HostApplicationBuilder builder = Host.CreateApplicationBuilder();
+
+// Configure logging
+builder.Logging.AddConsole();
+builder.Logging.SetMinimumLevel(LogLevel.Information);
+
+// Build a logger for startup configuration
+using ILoggerFactory loggerFactory = LoggerFactory.Create(loggingBuilder =>
+{
+    loggingBuilder.AddConsole();
+    loggingBuilder.SetMinimumLevel(LogLevel.Information);
+});
+ILogger<Program> logger = loggerFactory.CreateLogger<Program>();
+
+// Get environment variables for endpoint and taskhub with defaults
+string endpoint = Environment.GetEnvironmentVariable("ENDPOINT") ?? "http://localhost:8080";
+string taskHubName = Environment.GetEnvironmentVariable("TASKHUB") ?? "default";
+
+// Split the endpoint if it contains authentication info
+string hostAddress = endpoint;
+if (endpoint.Contains(';'))
+{
+    hostAddress = endpoint.Split(';')[0];
+}
+
+// Determine if we're connecting to the local emulator
+bool isLocalEmulator = endpoint == "http://localhost:8080";
+
+// Construct a proper connection string with authentication
+string connectionString;
+if (isLocalEmulator)
+{
+    // For local emulator, no authentication needed
+    connectionString = $"Endpoint={hostAddress};TaskHub={taskHubName};Authentication=None";
+    logger.LogInformation("Using local emulator with no authentication");
+}
+else
+{
+    // For Azure, use DefaultAzure authentication
+    connectionString = $"Endpoint={hostAddress};TaskHub={taskHubName};Authentication=DefaultAzure";
+    logger.LogInformation("Using Azure endpoint with DefaultAzure authentication");
+}
+
+logger.LogInformation("Using endpoint: {Endpoint}", endpoint);
+logger.LogInformation("Using task hub: {TaskHubName}", taskHubName);
+logger.LogInformation("Host address: {HostAddress}", hostAddress);
+logger.LogInformation("Connection string: {ConnectionString}", connectionString);
+logger.LogInformation("This worker implements a news article generator workflow with multiple specialized agents");
+
+// Create loggers for each service and activity
+var researchAgentServiceLogger = loggerFactory.CreateLogger<ResearchAgentService>();
```

Comment:
AFAIK, this *could* all be done via DI, which *should* simplify things quite a bit.